src/diffpy/utils/scattering_objects/diffraction_objects.py

```diff
@@ -364,99 +368,92 @@ def tth_to_q(self):
 
     def q_to_d(self):
         r"""
-        Helper function to convert q to d using :math:`d = \frac{2 \pi}{q}`
-
-        adds a small value (epsilon = 1e-10) to `q` where `q` is close to zero to avoid division by zero
+        Helper function to convert q to d using :math:`d = \frac{2 \pi}{q}`, set dmax to DMAX=100
 
         Parameters
         ----------
         q : array
-            An array of :math:`q` values
+            The array of :math:`q` values
 
         Returns
         -------
         d : array
-            An array of :math:`d` values in the inverse of the units of ``wavelength``
+            The array of :math:`d` values in the inverse of the units of ``wavelength``
         """
-        epsilon = 1e-10
         q = np.asarray(self.on_q[0])
-        q = np.where(np.abs(q) < epsilon, q + epsilon, q)
-        return (2 * np.pi) / q
+        d = np.where(q != 0, (2 * np.pi) / q, np.inf)
+        d = np.minimum(d, DMAX)
```

Comment:
This is not very readable.  Also, I guess it can end up with a bunch of values of D that are the same.  I think there is a better way to do this maybe.  Also, how does the user overrid DMAX if they want?

Reply:
The user can use insert_scattering_quantity to overrid or directly set the attributes. But I agree with you that this might be problematic. Also the appropriate limit seems a bit different for different wavelength. Maybe it's better to set d = 2 * previous_d if q = 0?
e.g. q = [0, 1] => d = [2pi, 4pi]
q = [0, 2] => d = [pi, 2pi]